packages/contracts/contracts/rbac.clar

```diff
@@ -0,0 +1,169 @@
+(define-constant CONTRACT_OWNER tx-sender)
+
+;; Stores roles granted to users
+;; Each user can have up to 128 roles ranging from 0-127.
+;; Roles are stored as bitmask values represented as uint's.
+(define-map UserRoles
+  {user: principal}
+  {roles: uint}
+)
+
+;; Stores the roles granted to permissions
+;; Each permission can be granted to up to 128 roles ranging from 0-127
```

Comment:
We should exclude 0, because that role is owned by all users by default, hence no access control. Users of the contract might get false impressions when they use `0`. At least make it very clear that `0` is granted to all users.

Reply:
By default users don't have any role (they are not stored in our map). And when we call `get-roles` function it will always return `u0`.
`u0` represented in binary will consist of 128 zeros. And it means that not a single role is assigned to that user.

If we assign a role to user and then revoke it, then in our map it will look like this:
```
{user: `SP123}
{roles: 0}
```
And once again, when we call `get-user` it will return `u0`.

`2 ^ 0 = 1` therefore `0` is a separate role.

Reply:
All users will always have role `u0`.

Reply:
I think we need to implement what you suggested [here](https://github.com/labs3/daoOS/pull/16#pullrequestreview-645499114) which is clear distinction between `roleId` and set of roles (let's call it for now `roleSet`) we store in `UserRoles` and `PermissionRoles` maps.

When we grant new role with any `roleId` that wasn't previously granted, we modify `roleSet` using following math equation: `roleSet = roleSet + (2 ^ roleId)`
And it should be interpreted as flipping `roleId` bit in `roleSet` from `0` to `1`. 

`roleId = u0` is our least significant bit in 128 bit uint stored as `roleSet` (first bit from right)
`roleId = u127` is our most significant bit in 128 bit uint stored as `roleSet` (first bit from left)



That being said, all users will have at the beginning `roleSet = u0` => all bits in this value are set to `0` which is equivalent of having no `roleId` assigned.